docker/dereferencer/dereferencer-py/dereferencer/derefEntity.py

```diff
@@ -0,0 +1,102 @@
+# This file will contain all the classes and functions needed to complete
+# the dereferencing process
+
+import logging
+from pathlib import Path
+import json
+import requests
+import os
+
+from .helpers import resolve_path
+
+log = logging.getLogger(__name__)
+
+
+def metadata_folder_from_config():
+    local_default = str(resolve_path("./memory_dereferencer", versus="dotenv"))
+    folder_name = os.getenv("METADATA_FILES_FOLDER", local_default)
+    return folder_name
+
+
+METADATA_FOLDER = metadata_folder_from_config()
+
+
+def check_metadata_folder_exists():
+    metadata_folder = metadata_folder_from_config()
+    if not os.path.exists(metadata_folder):
+        log.info(
+            f"metadata folder {metadata_folder} does not exist, creating it")
+        os.makedirs(metadata_folder)
+
+
+METADATA_FILE = METADATA_FOLDER + "/metadata.json"
+
+
+def make_metadata(input_list):
+    result = []
+
+    if input_list is None:
+        return result
+
+    for item in input_list:
+        if isinstance(item, dict):
+            for key, values in item.items():
+                if isinstance(values, list):
+                    children = make_metadata(values)
+                    result.append(
+                        {key: {"completed": False, "children": children}})
+                else:
+                    result.append({values: {"completed": False}})
+        elif isinstance(item, str):
+            result.append({item: {"completed": False}})
```

Comment:
Issue for dereference branch / will be converted into issue to keep track of convo. It can be discussed whether or not is can be stored in GraphDB or/and on local disk.